src/stream/src/common/log_store_impl/kv_log_store/reader.rs

```diff
@@ -149,6 +154,12 @@ impl<S: StateStore> LogReader for KvLogStoreReader<S> {
     }
 
     async fn next_item(&mut self) -> LogStoreResult<(u64, LogStoreReadItem)> {
+        while *self.is_paused.borrow_and_update() {
+            self.is_paused
+                .changed()
+                .await
+                .map_err(|_| anyhow!("unable to subscribe resume"))?;
+        }
```

Comment:
just confirmation, if it receives two resume mutations consecutively, the watcher's state won't change, right?

Reply:
Yes. The watch channel only retain the last sent value. 

https://docs.rs/tokio/latest/tokio/sync/watch/index.html